pandas/tests/indexing/test_coercion.py

```diff
@@ -13,6 +14,27 @@
 ###############################################################
 
 
+@pytest.fixture(autouse=True, scope='class')
+def check_comprehensiveness(request):
```

Comment:
where is this used?

Reply:
The ``autouse`` and ``scope`` args make it so that it is used by every class within the module

Reply:
oh, ok, cool. thanks for this patch. Nice work!

Reply:
@WillAyd This is neat.  I'm wondering: what would it take to put something like this together to check for permutations of arithmetic operations and operands?  I don't quite grok pytest's namespacing, in particular where `cls.klasses, cls.dtypes, cls.method` come from and what `request.node.session.items` corresponds to.

Reply:
`cls.klasses`, `cls.dtypes` and `cls.method` don't have anything to do with `pytest` - they are all class variables built into the tests in this module. I was inspired by [this link](https://docs.pytest.org/en/latest/example/special.html) on how to initially set this up, but had to tweak slightly given that link has a `session`-scoped fixture whereas here we are working with a class-level scope.

Basically `request.node.session.items` traverses from the fixture to the [node](https://docs.pytest.org/en/features/_modules/_pytest/nodes.html) (`pandas.tests.io.indexing.test_coercion::TestFoo`) and then goes from the node to the session. The session contains all of the test items, so iterating over them this code checks if all of the `klasses`, `dtypes` and `method` combinations set in the "in-scope" class are defined somewhere in the suite.

I don't entirely understand what you are trying to do with operations and operands but assuming you wanted to set up those combinations within a parametrization fixture I believe you could access that metadata by looking at `.callspec.params` on each object in `request.node.session.items`.

Reply:
> I don't entirely understand what you are trying to do with operations and operands

I going through #18824 I'm finding lots of cases that are not tested, saw this bit of code and thought it might be possible to enumerate e.g. `op = [__add__, __sub__, ...]`, `vec_classes = [Series, DatetimeIndex, np.ndarray, ...]`, `scalar_types = [...]`, `null_types=[...]`, `right = [...]` and check that all the cases are tested.  

That would also be helpful because there are a _ton_ of cases where tests are duplicated because `test_foo` and `test_bar` both test `foo+bar`, `bar+foo`. Not that this is a big problem, but it'd be nice to be systematic about it.